pkg/proto/schema_manager/loader.go

```diff
@@ -0,0 +1,152 @@
+package schema_manager
+
+import (
+	"context"
+	"fmt"
+	"strings"
+)
+
+import (
+	"github.com/arana-db/arana/pkg/mysql"
+	"github.com/arana-db/arana/pkg/proto"
+)
+
+const (
+	OrderByOrdinalPosition   = " ORDER BY ORDINAL_POSITION"
+	TableMetaDataNoOrder     = "SELECT TABLE_NAME, COLUMN_NAME, DATA_TYPE, COLUMN_KEY, EXTRA, COLLATION_NAME, ORDINAL_POSITION FROM information_schema.columns WHERE"
+	TableMetaDataSQL         = TableMetaDataNoOrder + OrderByOrdinalPosition
+	TableMetaDataSQLInTables = TableMetaDataNoOrder + " TABLE_NAME IN (%s)" + OrderByOrdinalPosition
+	IndexMetaDataSQL         = "SELECT TABLE_NAME, INDEX_NAME FROM information_schema.statistics WHERE TABLE_NAME IN (%s)"
+)
+
+func Load(db proto.DB, tables []string) map[string]TableMetadata {
+	var (
+		tableMetadataMap  = make(map[string]TableMetadata, len(tables))
+		indexMetadataMap  map[string][]IndexMetadata
+		columnMetaDataMap map[string][]ColumnMetadata
+	)
+	columnMetaDataMap = LoadColumnMetaDataMap(db, tables)
+	if columnMetaDataMap == nil {
+		indexMetadataMap = nil
+	} else {
+		indexMetadataMap = LoadIndexMetaData(db, tables)
+	}
```

Comment:
How about use `if columnMetaDataMap != nil {...}` instead? I think the `indexMetadataMap` is nil by default already.